### PR TITLE
infection modulation

### PR DESCRIPTION
**Pull request by tozanski**

### CODE REVIEW

File: julia/src/Simulation/params/modulations.jl [hunk 1]
```diff
@@ -0,0 +1,33 @@
+struct TanhModulation
+  weight_detected::Float64
+  weight_deaths::Float64
+  loc::Float64
+  scale::Float64
+  limit_value::Float64
+  TanhModulation(;weight_detected::Real, weight_deaths::Real, loc::Real, scale::Real, limit_value::Real) = new(weight_detected, weight_deaths, loc, scale, limit_value)
```
Comment (eMaerthin) [inline julia/src/Simulation/params/modulations.jl:7]: what means ";" in front of list of arguments?

File: julia/src/Simulation/params/modulations.jl [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+end
+
+function (f::TanhModulation)(state::SimState, params::SimParams, event::Event)
+  @assert kind(event) == TransmissionEvent
+
+  ck = contactkind(event)
+  if ConstantKernelContact !== ck && SporadicContact !== ck && FriendshipContact !== ck
```
Comment (eMaerthin) [inline julia/src/Simulation/params/modulations.jl:14]: maybe there could be different functions for different kernels in general? say the constant kernel is more affected than friendship kernel by reduction of contacts?

File: julia/src/Simulation/params/modulations.jl (hunk 1 above)
Comment (tozanski) [reply julia/src/Simulation/params/modulations.jl:7]: it separates positional arguments from keyword arguments. There are only keyword arguments in this constructor

File: julia/src/Simulation/params/modulations.jl (hunk 2 above)
Comment (tozanski) [reply julia/src/Simulation/params/modulations.jl:14]: This is one of possible functions, anyone is free to define a function that is more specialized and behaves differently for different kernels

**Pull request merged**